### PR TITLE
Move MockOptions into its own compilation unit.

### PR DESCRIPTION
**Pull request by oschaaf**
Part of an effort to attempt to reduce memory usage during
the build, to address ASAN flakes in CI.

Signed-off-by: Otto van der Schaaf <oschaaf@we-amp.com>

### CODE REVIEW

**Comment by oschaaf** [comment]
(cross referencing associated issue: #272)

**Comment by mum4k** [comment]
@dubious90 please review and assign back to me when ready.

File: include/nighthawk/client/options.h [hunk 1]
```diff
@@ -9,6 +9,8 @@
 #include "envoy/config/cluster/v3/cluster.pb.h"
 #include "envoy/config/core/v3/base.pb.h"
 
+#include "nighthawk/common/termination_predicate.h"
```
Comment (dubious90) [inline include/nighthawk/client/options.h]: A little surprised to see this include with no other corresponding changes.  Why was this suddenly necessary?

Comment (oschaaf) [reply include/nighthawk/client/options.h]: this is an include-what-you-use fix; the need for which became apparent when isolating the compilation unit. Apparently, before doing that, the include leaded into scope via another include (the original mock_options.h) in a test which needed it.

**Pull request merged**